### DISCUSSION TITLE
Supporting Generation 2 Virtual Machines in Azure

### DISCUSSION THREAD

**Discussion by ifutant** [actions/virtual-environments#6780] [Ideas]
Hello! 

I am currently consuming this repo to create images for our private runners and it's pretty awesome!
I have plans to prototype migrating from V1 to V2 VM generations. 
I created a PR to this repo to facilitate the modification of the image_sku.
I am curious if any other users are interested in using V2 Azure virtual machines. 

[Pending Pull Request](https://github.com/actions/runner-images/pull/6756)

My main desire to deploy gen 2 vms: 
[Trusted Launch](https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch)

Specific components/features that intrigue:
- Trusted Launch, vTPM, secure boot, faster boot time compared to gen1.

Additional Reading:
[Support for generation 2 VMs on Azure](https://learn.microsoft.com/en-us/azure/virtual-machines/generation-2)
[Should I create a generation 1 or 2 virtual machine in Hyper-V?](https://learn.microsoft.com/en-us/windows-server/virtualization/hyper-v/plan/should-i-create-a-generation-1-or-2-virtual-machine-in-hyper-v)

**Comment by ifutant** [comment]
Anyone interested in this functionality?

**Comment by bartwitkowski** [comment]
Hi,
This question is almost 2y old.
Have something changed during that time?
I cannot spot any configuration to build Gen 2 OS images.
Azure Defender recommend using Trusted Launch (vTPM, SecureBoot).

**Reply by aharden** [reply]
Because we are managing our VM images in an Azure Compute Gallery, we maintain our own derivative of the `runner-images` Packer templates with a `source` block that is appropriate for our use.  Part of our customization is the use of Gen 2 VM base images, which work fine.  We use the `runner-images` templates' `build` blocks with our own custom tool needs added.

**Reply by PetrPospisil** [reply]
https://github.com/actions/runner-images/pull/13543
